### PR TITLE
SMB: add SessionSetup responses

### PR DESCRIPTION
**Pull request by gpotter2**
- adds answers to SessionSetup requests -> NTLM challenge

This needs https://github.com/ivre/masscanned/pull/45 to be able to work in prod / add a functional test :P

### CODE REVIEW

**Comment by p-l-** [comment]
@gpotter2 is this ready for review now?

**Comment by gpotter2** [comment]
It is, apart from the fact that I should probably add a functional test :p

**Comment by p-l-** [comment]
Also, can you rebase against current master?

File: src/proto/smb.rs [hunk 1]
```diff
@@ -922,35 +1067,48 @@ impl MPacket for SMB2SetupRequest {
...
+        if !matches!(self.d.state, SMB2SetupRequestState::End) {
+            return None;
+        }
+        let mut resp: Vec<u8> = Vec::new();
+        resp.extend_from_slice(&0x9_u16.to_le_bytes()); // StructureSize
+        resp.extend_from_slice(&0x0_u16.to_le_bytes()); // SessionFlags
+        resp.extend_from_slice(&0x48_u16.to_le_bytes()); // SecurityBufferOffset
+        resp.extend_from_slice(&(SECURITY_BLOB_CHALLENGE.len() as u16).to_le_bytes()); // SecurityLen
+        resp.extend_from_slice(SECURITY_BLOB_CHALLENGE); // SecurityBlob
+        warn!("SMB2 SessionSetup-Reply");
```
Comment (Frky) [inline src/proto/smb.rs:1090]: TODO: use log module (but can be done in a future commit)

**Review by Frky — APPROVED** [review]
Thanks for the PR.

**Pull request merged**